### ISSUE TITLE
pmap behavior with err_stop = true

### ISSUE THREAD

**Issue by simonster** [JuliaLang/julia#7808]
Right now, the code:

```julia
pmap((x)->(sleep(x); println(x); error("error")), 1:2; err_stop=true)
```

gives output

```
        From worker 2:  1
exception on 2: ERROR: error
 in error at error.jl:21
 in anonymous at none:1
 in anonymous at multi.jl:855
 in run_work_thunk at multi.jl:621
 in anonymous at task.jl:855
        From worker 3:  2
exception on 3: ERROR: error
 in error at error.jl:21
 in anonymous at none:1
 in anonymous at multi.jl:855
 in run_work_thunk at multi.jl:621
 in anonymous at task.jl:855
2-element Array{Any,1}:
 ErrorException("error")
 ErrorException("error")
```

If a worker throws an error, maybe `pmap` should interupt the other workers immediately rather than waiting for them to finish their jobs? Also, if `err_stop = true`, maybe the error should be rethrown?

**Comment by samoconnor** [comment]
See fix here #12943

**Comment by amitmurthy** [comment]
`err_stop` has been replaced with `on_error` and `retry_n` keyword args in 0.5
